### ISSUE TITLE
Stochastic predictions with GraphSAGE

### ISSUE THREAD

**Issue by youph** [williamleif/GraphSAGE#63]
During training, each graphsage layer takes uniform samples from each node's neighbours, a new sample at each training iteration. With a large number of samples taken during the training, the entire neighbourhood of each node is (probably) covered, and the model learns node embeddings from node own features and the aggregated features of essentially entire node neighbourhoods.

But what about predictions? If at prediction time you again take random samples of a node's neighbours, then repeated predictions by the same model on the same node will give different results (due to different node neighbours sampled each time - see Fig. 1 of the paper), i.e., predictions will be stochastic. Do you observe this stochasticity of predictions? If not, how do you sample neighbours at prediction time?

**Comment by RexYing** [comment]
Yes there will be stochasticity. With ~10 and 25 neighbors at each layer we do not observe too much variability.
Also sample can be performed at training time only. At inference time, if efficiency is not a concern you can aggregate the entire neighborhood as well, to get deterministic results.
